File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:
long long int binarySearch(int n){
        int s = 0;
        int e = n;
        long long int ans = -1;
        long long int mid = s + (e-s)/2;
              
        while(s<=e){
            long long int multi = mid * mid;  
            if(multi == n) 
                return mid;
                
            if(multi < n){
               ans=mid;
               s=mid+1;
            }
            else{
            e = mid - 1; 
            }
        mid = s+(e-s)/2;
    }
    return ans;
    }
 int mySqrt(int x) {
        return binarySearch(x);
 }
};

File: 0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        start = 0
        end = x
        mid = start +(end-start)//2
        ans=-1
        while start<=end:
            multi = mid * mid
            if multi == x:
                return mid
            elif multi < x:
                ans=mid
                start = mid+1
            else:
                end=mid-1
            mid=start+(end-start)//2
        return ans